### ISSUE TITLE
Some game cause the Apple IIe to hang

### ISSUE THREAD

**Issue by frozen-signal** [frozen-signal/Apple_IIe_MMU_IOU#41]
Some games are not working when using a specific setup. Not exactly sure what could be the problem. Possibly another timing issue.
This setup works without problem:
- 6502 with un-enhanced ROMs
- 80-Col Expansion card (820-0067-D)
- Apple 5.25 Drive Controller Card (655-0101-E)

But using instead a 65C02 with enhanced ROMs will cause problems with Doulbe HiRes games. For example, Aliens will hang when switching to the graphic mode.
Seems to be tied to the 80-Col card, but it's possible to re-create the problem without the card present. For example, Aliens normally can run without the 80-Col card. But will still hang without it.

Apple IIe Diagnostic v2.1 also reports problems with the 80-Col card.

**Comment by frozen-signal** [comment]
There seems to be a problem with CASEN_N and EN80_N.

This is a logic analyzer capture of the game ALIENS, with a trigger on the falling edge of EN80_N:
Both captures are without a 80-Col Card present, on an enhanced Apple IIe.
With the ASIC MMU:
![image](https://github.com/frozen-signal/Apple_IIe_MMU_IOU/assets/132361725/47b74ac2-c93e-47fc-ac85-9ff065ea373a)
With the MMU from this repository:
![image](https://github.com/frozen-signal/Apple_IIe_MMU_IOU/assets/132361725/1b11f99d-77a2-4906-8384-ded29c340ff4)

For some reason, the CASEN_N and EN80_N signals are wrong in the prototype.

**Comment by frozen-signal** [comment]
Fixed in https://github.com/frozen-signal/Apple_IIe_MMU_IOU/commit/5f5cba5f23c98e7ff753abff4ef4aa12a678f67c

At first, I though that the SELMB_N computation of the emulator schematics was incorrect. So, I re-wrote SELMB_N to be closer to the ASIC schematics. This new implementation is much simpler than the other schematics one, but do not fix the problem described above.

Later, I was convinced that PG2 somehow ended up with the wrong value and started searching through the logic analyzer traces until I found this:
![image](https://github.com/frozen-signal/Apple_IIe_MMU_IOU/assets/132361725/6307d9eb-f37c-4d62-9a69-ab13a215fccb)
The CPU places $C054 on the address bus to reset the soft-switch PG2, But while the address is crumbling, the MMU momentarily sees $C055 and sets it back.

To fix, the latch enabling that control these soft-switches must be forced HIGH during PHI_1 (i.e. during PHI_0 LOW).


This fixes the problem with ALIENS, Apple IIe Diagnostic v2.1, as well as almost every game. Except "The Black Cauldron" that seems to have an problem. Will open a different issue for that one.
